controllers/flinkcluster/flinkcluster_converter.go

```diff
@@ -45,18 +46,20 @@ import (
 // underlying Kubernetes resource specs.
 
 const (
-	preStopSleepSeconds     = 30
-	flinkConfigMapPath      = "/opt/flink/conf"
-	flinkConfigMapVolume    = "flink-config-volume"
-	submitJobScriptPath     = "/opt/flink-operator/submit-job.sh"
-	gcpServiceAccountVolume = "gcp-service-account-volume"
-	hadoopConfigVolume      = "hadoop-config-volume"
-	jobManagerAddrEnvVar    = "FLINK_JM_ADDR"
-	jobJarUriEnvVar         = "FLINK_JOB_JAR_URI"
-	jobPyFileUriEnvVar      = "FLINK_JOB_PY_FILE_URI"
-	jobPyFilesUriEnvVar     = "FLINK_JOB_PY_FILES_URI"
-	hadoopConfDirEnvVar     = "HADOOP_CONF_DIR"
-	gacEnvVar               = "GOOGLE_APPLICATION_CREDENTIALS"
+	preStopSleepSeconds           = 30
+	flinkConfigMapPath            = "/opt/flink/conf"
+	flinkConfigMapVolume          = "flink-config-volume"
+	submitJobScriptPath           = "/opt/flink-operator/submit-job.sh"
+	gcpServiceAccountVolume       = "gcp-service-account-volume"
+	hadoopConfigVolume            = "hadoop-config-volume"
+	jobManagerAddrEnvVar          = "FLINK_JM_ADDR"
+	jobJarUriEnvVar               = "FLINK_JOB_JAR_URI"
+	jobPyFileUriEnvVar            = "FLINK_JOB_PY_FILE_URI"
+	jobPyFilesUriEnvVar           = "FLINK_JOB_PY_FILES_URI"
+	hadoopConfDirEnvVar           = "HADOOP_CONF_DIR"
+	gacEnvVar                     = "GOOGLE_APPLICATION_CREDENTIALS"
+	maxUnavailableDefault         = "0%"
+	maxUnavailableApplicationMode = "50%"
```

Comment:
@kassybas what are you thinking around this? why do we need to set it differently for app mode?